src/posts/2020-08-20-an-indepth-guide-to-aria-roles.md

```diff
@@ -0,0 +1,190 @@
+---
+layout: post
+title: An in-depth guide to ARIA roles
+description: ARIA roles are one half of the predefined categories of attributes used to describe elements that may not exist natively in browsers or may not be understood by screen readers and other assistive technologies.
+category: Background
+author: Duncan Jimbo
+date: 2020-08-20
+eleventyNavigation:
+  key: {{ category }}
+  title: {{ title }}
+tags:
+  - background
+---
+
+The ARIA specification is divided into categories of attributes, one of which is Roles, and the other being States & Properties. ARIA roles, which are covered in this article, are generally used to describe elements that may not exist within HTML, or elements which may not have full cross-browser support, or have implementation gaps in screen readers and other assistive technologies. For instance, ARIA roles can be useful in situations where native HTML semantics are not understood in legacy user agents (e.g., browsers).
+
+## Usage
+
+ARIA roles are added to elements using the `role="<ROLE_TYPE>` attribute, where `ROLE_TYPE` represents the specific role being added. Once an ARIA role is set on an element, it should not be changed, regardless of the state of the element. In some cases, roles have to be paired with the appropriate ARIA state or property. However, this article's examples have reduced markup to limit the scope to just ARIA roles.
+
+ARIA roles are broken down into six categories:
+
+1. Abstract Roles
+2. Document Structure Roles
+3. Landmark Roles
+4. Live Region Roles
+5. Widget Roles
+6. Window Roles
+
+### Abstract roles
+
+Abstract roles are only intended for use by browsers to help organize and streamline a document. They should not be used by developers writing HTML markup. Doing so will not result in any meaningful information being conveyed to assistive technologies or to users.
+
+### Document Structure roles
+
+Document roles are used to provide a structural description for a section of content. A majority of the available roles in this category resemble available semantic HTML tags and it is recommended to only use them when an appropriate native tag is not available. The roles in this category are also not typically interactive in nature, but may be used to contain other roles that are interactive. The `toolbar"` and `tooltip"` roles, demonstrated below, serve as good examples of when to use Document Structure roles to provide information to assistive technologies when the equivalent native HTML tag is not available.
```

Comment:
>The roles in this category are also not typically interactive in nature, 

New thought, new paragraph. Shorter paragraphs will help the reader stay engaged, expecially on a longer and more technical piece like this.